core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -65,12 +72,20 @@ public TermSuggestionBuilder(String name) {
      * tokens in the suggest text.
      * </ol>
      */
-    public TermSuggestionBuilder suggestMode(String suggestMode) {
+    public TermSuggestionBuilder suggestMode(@Nullable SuggestMode suggestMode) {
```

Comment:
I commented above on defaults (https://github.com/elastic/elasticsearch/pull/16278/files#r51552740) and I think where we have defaults we should make sure they cannot be nulled as this will create hard to debug errors for the user. IMO we should stick to the mantra of not allowing an invalid builder object to be built in the Java API so that the user can easily see where they have made errors in their own code.